deploy/test/01_deploy_ens_contracts.js

```diff
@@ -0,0 +1,48 @@
+const { ethers } = require('hardhat');
+
+module.exports = async ({getNamedAccounts, deployments, network}) => {
+  const {deploy} = deployments;
+  const {deployer} = await getNamedAccounts();
+  
+  // Only deploy mock ENS contracts on test networks
+  if (!network.tags.test) {
+    return;
+  }
+  
+  // Deploy MockENSRegistry
+  await deploy('ENSRegistry', {
+    from: deployer,
+    contract: 'MockENSRegistry',
+    args: [],
+    log: true,
+  });
+  
+  const registry = await ethers.getContract('ENSRegistry');
+  
+  // Deploy MockReverseRegistrar
+  await deploy('ReverseRegistrar', {
+    from: deployer,
+    contract: 'MockReverseRegistrar',
+    args: [registry.address],
+    log: true,
+  });
+  
+  // Deploy MockPublicResolver
+  await deploy('PublicResolver', {
+    from: deployer,
+    contract: 'MockPublicResolver',
+    args: [registry.address],
+    log: true,
+  });
```

Comment:
Refactor to use the real `ENSRegistry`, `ReverseRegistrar` and `PublicResolver` contracts if possible. If it's not possible or would introduce a great deal of extra code, stop for feedback and let me know the problem in specific terms.